agent/consul/rpc_test.go

```diff
@@ -1682,23 +1683,98 @@ func getFirstSubscribeEventOrError(conn *grpc.ClientConn, req *pbsubscribe.Subsc
 	return event, nil
 }
 
-// assertBlockingQueryWakeupCount is used to assist in assertions for
-// blockingQuery RPC tests involving the two sentinel errors errNotFound and
-// errNotChanged.
-//
-// Those tests are a bit racy because of the timing of the two goroutines, so
-// we relax the check for the count to be within a small range.
-//
-// The blocking query is going to wake up every interval, so use the elapsed test
-// time with that known timing value to gauge how many legit wakeups should
-// happen and then pad it out a smidge.
-func assertBlockingQueryWakeupCount(t testing.TB, interval time.Duration, start time.Time, gotCount int) {
+// channelCallRPC lets you execute an RPC async. Helpful in some
```

Comment:
Start here

Reply:
This is a blocking query that cannot be canceled (thanks `net/rpc`). By wrapping it in a 1-buffered channel we can effectively orphan it randomly in the test and it will eventually be GCed (since it can write into the channel and then die) and if it does randomly conclude it isn't able to interact with the original caller's state. So it's a waste of a bit of ram during the test, but it's just dark matter in the system now.